src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -305,9 +307,11 @@ private void WatchForFileSystemEventsThreadStart(object arg)
 
                 for (long i = 0; i < numEvents.ToInt32(); i++)
                 {
+                    Debug.Assert(eventPaths[i][eventPaths[i].Length - 1] != '/');
```

Comment:
Similar to above, if it's possible eventPaths[i].Length could be 0, this assert needs to be fixed, and if it's not possible, consider adding another assert before this that does a length check.  Also, consider adding messages to the asserts; since we don't have line number information in failed asserts on unix, messages help to distinguish them.